### PR TITLE
Dont link to dashboard from footer

### LINKED ISSUE #6137: List only built versions on the footer
When a version is `built=False` our code links to the version detail page

https://github.com/readthedocs/readthedocs.org/blob/89a58a07716ccf5793e0a241be174d52d1f03c76/readthedocs/builds/models.py#L296-L303

This method gets called from `sort_version_aware`, here we are passing all active versions (including not built).

https://github.com/readthedocs/readthedocs.org/blob/89a58a07716ccf5793e0a241be174d52d1f03c76/readthedocs/projects/models.py#L953-L986

This is a problem in .com, where we don't have access to that `URLCONF`, so reverse doesn't work.

I'm thinking two options:

1. Only list built versions, this is modify this method and add a filter (this method is only used in the footer).

https://github.com/readthedocs/readthedocs.org/blob/89a58a07716ccf5793e0a241be174d52d1f03c76/readthedocs/projects/models.py#L953

2. Only change this in .com, adding a new kwarg to .public (`only_builts=False`)

I'm more inclined to 1, so we affect .org and .com,  giving a link to the dashboard to a user that doesn't have permissions doesn't look correct.

### PR DESCRIPTION
**Pull request by stsewd**
We use the `get_absolute_url` in other parts of the code
where it makes sense to link to the dashboard.
So I'm adding an additional parameter.

Closes #6137

### CODE REVIEW

File: readthedocs/rtd_tests/tests/test_footer.py [hunk 1]
```diff
@@ -235,11 +240,12 @@ class TestFooterPerformance(APITestCase):
 
     # The expected number of queries for generating the footer
     # This shouldn't increase unless we modify the footer API
-    EXPECTED_QUERIES = 9
+    EXPECTED_QUERIES = 13
```
Comment (ericholscher) [inline readthedocs/rtd_tests/tests/test_footer.py]: Hmm, this is less than ideal. Is this an additional query for each version?

Comment (stsewd) [reply readthedocs/rtd_tests/tests/test_footer.py]: The explanation is the commit

> We were testing with non-build versions.
> But in production is more common to hit this with build versions.

Comment (stsewd) [reply readthedocs/rtd_tests/tests/test_footer.py]: That number is the "normal" number of queries, we were testing for a no common case where the version wasn't built

Comment (ericholscher) [reply readthedocs/rtd_tests/tests/test_footer.py]: Ah I see, it's in the version_compare data. I wonder if we should be caching this, seems simple enough to cache if it's a lot of our footer queries.

Comment (ericholscher) [reply readthedocs/rtd_tests/tests/test_footer.py]: I will _never_ look at commit messages in PR's, so it's probably better to put it in the PR description.

**Pull request merged**